### DISCUSSION TITLE
Current features in work, ideas of new features and issues available to be taken into work

### DISCUSSION THREAD

**Discussion by drfenixion** [drfenixion/freecad.robotcad#52] [Ideas]
### Tasks:
@drfenixion work list:

- [x] 1. Sensors feature.
- [x] 2. Transfer all to ROS2 Jazzy.
- [x] 3. Issues fixing - Current

@Jsevenh work list:
1. SDF generating of Robot enviroment. - Current

### Ideas of new feature:
1. Integrate RL tools to result generated ROS2 project for scaffold of RL.
2. Integrate visual programming (flow programming). Maybe something like https://github.com/eProsima/node-red-ros2-plugin
4. Your idea. Be free to add ideas.

Here you can notice and discuss if you want to implement some feature.

**Comment by Jsevenh** [comment]
great

**Comment by Jsevenh** [comment]
Hello, I saw that you merged the sensors PR. I'm taking a short break from programming to recover from a bit of burnout, so I haven’t reviewed the code yet. I plan to dive into it and resume working on SDF starting January 9th. Wishing you a Happy New Year!

**Reply by drfenixion** [reply]
No problem, take your time. Happy New Year!

**Comment by Jsevenh** [comment]
What was your main reason for adding this  attributes with the prefix meta to all  elements? in the sdf_schema_parser, I realize it has some additional overhead when working with the sdf files , as they need to be removed

**Reply by drfenixion** [reply]
There is autoremover of meta attrs https://github.com/drfenixion/freecad.robotcad/blob/main/freecad/cross/sdf/sdf_parser/sdf_tree.py#L73

meta attrs used in https://github.com/drfenixion/freecad.robotcad/blob/main/freecad/cross/sensors/sensor_proxy.py#L317

**Comment by Jsevenh** [comment]
also on the need for   skipping elements  that have the include tag [here](https://github.com/drfenixion/freecad.robotcad/blob/main/freecad/cross/sdf/sdf_parser/sdf_schema_parser.py#L139C9-L141C73) having a pass statement ,makes more sense  so as to allow only a single file to be selected at a time and reduce the overhead of having to recursively find a required element ,

**Reply by drfenixion** [reply]
SDF schema is constructed on include tags. I will not work correct without it.

Include tags includes fragments of document and without that fragments element will be broken. F.e. https://github.com/gazebosim/sdformat/blob/181c0ef9f33a56f118ab8f185178c12ddd8b0e1f/sdf/1.11/collision.sdf#L31

**Reply by Jsevenh** [reply]
> SDF schema is constructed on include tags. I will not work correct without it.

I completely agree; appending them manually reduces the overhead. It’s much simpler to add them manually because, for instance, each link's include statement often imports unnecessary sub-elements. Some include tags even reference other include tags, leading to deeply nested structures and importing many unnecessary elements, which increases the overhead.

**Comment by Jsevenh** [comment]
Either way, I'll add a default parameter to control whether elements with "include" tags should be appended, setting it to true by default.

**Comment by Jsevenh** [comment]
Hello , it's been a while since I last checked in. I'm impressed with how much the Workbench has evolved—fantastic work @drfenixion  ,With some free time on my hands, I've started tackling the clutter in my command list. For the past few days, I've been grouping everything into logical sets. I've just made a [pull request](https://github.com/drfenixion/freecad.robotcad/pull/103) ,please do  make recommendations

**Comment by Jsevenh** [comment]
Regarding my ongoing work on the "SDF and Environment" integration, I've noted the new release which handles environment features. As this likely overlaps with my objectives, I will need to reassess my implementation targets.
 Consequently, I will be closing this pull request [#50](https://github.com/drfenixion/freecad.robotcad/pull/50).
Moving forward, I am considering a new approach: to hide all robot properties from the standard FreeCAD Data tab and develop a dedicated dockable widget for property editing. This should provide a more streamlined and user-friendly experience than the default FreeCAD property editor.'m thinking of starting with smaller, incremental changes to the workbench. I'm hopeful this approach will hopefully integrate useful features. What are your thoughts on this direction?

**Comment by drfenixion** [comment]
@Jsevenh - "develop a dedicated dockable widget for property editing" - It can be a lot of work with no obvious advantages. 
1) There are a lot of various propreties of varios RobotCAD etities.
2) FreeCAD users used to use the standart FreeCAD data properties style.

In my opinion at that stage of developing is better to implement some new feature.
